### ISSUE TITLE
Building OpenSSL 3.0.5 for i386 Darwin with `no-modules` causes duplicate symbol errors

### ISSUE THREAD

**Issue by CodingMarkus** [openssl/openssl#18950]
Building OpenSSL 3.0.5 for i386 Darwin with `no-modules` causes duplicate symbol errors:

```
duplicate symbol '_bn_div_words' in:
    crypto/bn/libcrypto-shlib-bn-586.o
    providers/liblegacy.a(liblegacy-lib-bn-586.o)
duplicate symbol '_bn_sub_part_words' in:
    crypto/bn/libcrypto-shlib-bn-586.o
    providers/liblegacy.a(liblegacy-lib-bn-586.o)
duplicate symbol '_bn_sqr_words' in:
    crypto/bn/libcrypto-shlib-bn-586.o
    providers/liblegacy.a(liblegacy-lib-bn-586.o)
duplicate symbol '_bn_mul_words' in:
    crypto/bn/libcrypto-shlib-bn-586.o
    providers/liblegacy.a(liblegacy-lib-bn-586.o)
duplicate symbol '_bn_add_words' in:
    crypto/bn/libcrypto-shlib-bn-586.o
    providers/liblegacy.a(liblegacy-lib-bn-586.o)
duplicate symbol '_bn_mul_add_words' in:
    crypto/bn/libcrypto-shlib-bn-586.o
    providers/liblegacy.a(liblegacy-lib-bn-586.o)
duplicate symbol '_bn_sub_words' in:
    crypto/bn/libcrypto-shlib-bn-586.o
    providers/liblegacy.a(liblegacy-lib-bn-586.o)
ld: 7 duplicate symbols for architecture i386
```

Configure was created using `./Configure darwin-i386-cc no-module`. 

Works fine for `darwin64-x86_64-cc` and `darwin64-arm64-cc`. Also works fine for `iphone-cross` regardless of CPU (armv7, armv7s, arm64).

**Comment by CodingMarkus** [comment]
Just figured out: As a work-around, adding `no-shared` as well as `no-module` seems to fix that issue. 

But doesn't `no-module` imply that no shared libraries are created?

**Comment by mattcaswell** [comment]
> But doesn't no-module imply that no shared libraries are created?

No. This is just about dynamically loadable modules such as engines and providers.

**Comment by prasad1240** [comment]
@CodingMarkus 
Is duplicate symbol errors issue resolved after you added "no-module" & "no-shared"?

Still, I am facing an issue with the configuration 
`./Configure ios64-xcrun no-tests no-shared no-module`
If it is working then please let me know what is your configuration.

**Comment by CodingMarkus** [comment]
> Is duplicate symbol errors issue resolved after you added "no-module" & "no-shared"?

Yes, `no-shared` + `no-module` fixed that issue for me.

I've never built `ios64-xcrun`, though. I built `iphone-cross` which builds armv7, armv7s, and arm64 versions and that worked fine when just using `no-modules`. I didn't specify any configuration other than 
`./Configure iphone-cross no-module`.

**Comment by t8m** [comment]
Do you still see this issue with recent releases?

**Comment by CodingMarkus** [comment]
I don't have the same setup available anymore. 

- My system is now macOS 14 (Sonoma)
- The clang version in Xcode has been updated to `clang-1403.0.22.14.1`

And with that setup I cannot build i386 at all, as the SDK no longer provides symbols for i386, so I end up with tons of `Undefined symbols for architecture i386` errors during linking phase. And older SDKs won't work on Sonoma anymore (actually the one I use only works thanks to some patching of Xcode 14.3.1, since Xcode 15 causes plenty of issues in my projects that I cannot easily fix).

But I can try to build it on an older build machine with an older SDK if there is a reason to assume that this issue has been fixed in the meantime.

**Comment by nhorman** [comment]
marking as inactive, will be closed at the end of 3.4 dev barring further input
